sr/comp/cli/show_match_scores.py

```diff
@@ -0,0 +1,233 @@
+    ranking: Mapping[TLA, Optional[LeaguePosition]]
+
+    if match.type == MatchType.knockout:
+        score_data = comp.scores.knockout
+    elif match.type == MatchType.tiebreaker:
+        score_data = comp.scores.tiebreaker
+    elif match.type == MatchType.league:
+        score_data = comp.scores.league
+
+    match_id = (match.arena, match.num)
+
+    if match_id in score_data.game_points:
+        league_points = score_data.ranked_points[match_id]
+        game_points = score_data.game_points[match_id]
+        ranking = degroup(score_data.game_positions[match_id])
+    else:
+        league_points = {}
+        game_points = {}
+        ranking = {}
+        for team in match.teams:
+            if team is not None:
+                league_points[team] = None
+                game_points[team] = None
+                ranking[team] = None
+
+    corner_data: Dict[int, MatchCorner] = {}
+
+    for corner, team in enumerate(match.teams):
+        match_id = (match.arena, match.num)
+
+        if team:  # corner occupied
+            corner_data[corner] = MatchCorner(
+                tla=team,
+                ranking=ranking[team],
+                game=game_points[team],
+                league=league_points[team],
+            )
+        else:
+            corner_data[corner] = MatchCorner(
+                tla=None,
+                ranking=None,
+                game=None,
+                league=None,
+            )
+
+    return MatchResult(
+        num=match.num,
+        arena=match.arena,
+        display_name=match.display_name,
+        corners=corner_data,
+    )
+
+
+def match_index(matches: List[MatchSlot], match_num: int) -> int:
+    "Returns the index of the first slot that contains the given match number"
+    for idx, slots in enumerate(matches):
+        for match in slots.values():
+            if match.num == match_num:
+                return idx
+
+    # if no match is found use the last one
+    return idx
+
+
+def generate_displayed_headings(num_corners: int) -> List[str]:
+    displayed_heading = ["Match"]
+
+    for _ in range(num_corners):
+        displayed_heading.append("Zone")
+        displayed_heading.append("TLA")
+        displayed_heading.append("Rank")
+        displayed_heading.append("Game")
+        displayed_heading.append("League")
+
+    return displayed_heading
+
+
+def generate_displayed_match(match: MatchResult, num_corners: int) -> List[List[str]]:
+    displayed_match = []
+    displayed_corners = []
+
+    for zone, (tla, ranking, game, league) in match.corners.items():
+        displayed_corner: List[str] = []
+
+        displayed_corner.append(str(zone))
+        if tla is not None:
+            displayed_corner.append(tla)
+            displayed_corner.append("??" if ranking is None else str(ranking))
+            displayed_corner.append("??" if game is None else str(game))
+            displayed_corner.append("??" if league is None else str(league))
```

Comment:
I suggest maybe using `-` rather than `??` for unknown score values? It still conveys the unknown nature while being less attention grabbing. `???` should still be used for unknown TLAs though as that's an existing SRComp standard (and is chosen to both visually look like a TLA and indicate the unknowable nature of the value).